.travis.yml

```diff
@@ -33,15 +33,22 @@ matrix:
 #            - BREW_INSTALL=python3
 
 install:
-  - pip install tox
+  - pip install flake8 tox
 #  - |
 #    if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
 #      if [[ -n "$BREW_INSTALL" ]]; then
 #        brew update
 #        brew install "$BREW_INSTALL"
 #      fi
 #    fi
-#    pip install tox
+
+before_script:
+  # stop the build if there are Python syntax errors or undefined names
+  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+  if [[ $TOXENV == py37 ]]; then
+    flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics ;
+    flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics ;
```

Comment:
This one I'm not in favor of, at least not yet because it is too strict.  But I like the idea of creating a new Issue to clean up some of the ``flake8`` warnings.

Reply:
Ok, as long as it is only producing warnings, I'm OK with this raising visibility for developers.  Thanks for pointing out that this won't fail the build.